File: solutions/typescript/rna-transcription/1/rna-transcription.ts
export function toRna(dna: string): string {
  return [...dna].map(c => toComplement(c)).join("")
}

function toComplement(n: string): string {
  switch (n) {
    case 'A':
      return 'U';
    case 'C':
      return 'G';
    case 'G':
      return 'C'
    case 'T':
      return 'A'
    default:
      throw 'Invalid input DNA.'
  }
}

File: solutions/julia/rna-transcription/1/rna-transcription.jl
function to_rna(dna)
    join(map(t, dna))
end

function t(c::Char)
    if c == 'G'
        'C'
    elseif c == 'C'
        'G'
    elseif c == 'T'
        'A'
    elseif c == 'A'
        'U'
    else
        throw(ErrorException(""))
    end
end

